### ISSUE TITLE
Evaluate rewiremock for mocking in JS tests

### ISSUE THREAD

**Issue by philippotto** [scalableminds/webknossos#2993]
rewiremock promises to be "not complex" and to be "the most powerful mocking library". Additionally, it offers type safety with flow in combination with dynamic import syntax. This would help a great deal, since even for flow-enabled test files, typing information is lost for modules which are required via mock-require.

https://www.npmjs.com/package/rewiremock

**Comment by daniel-wer** [comment]
After looking at this for an hour or so, I'm not convinced. The docs are very hard to follow :/
I couldn't even get the hoisting to work (the babel plugin does nothing for me?!), which would have been an advantage over our old mocking library.
The syntax to keep the flow types does look very complicated to me as well.

@philippotto I would suggest you to have a look at this as well maybe and then we'll decide. By the way, did you ever consider jest as a mocking library? It has much better docs, supports hoisting as well and at least from this issue (https://github.com/facebook/jest/issues/4133) it looks like flow types are not completely lost?

**Comment by philippotto** [comment]
For the record: rewiremock didn't work out for me, either, and since jest would need to replace ava, we agreed to stick with the current set up until something more promising comes along.
